secrets/resource_secret_scope.go

```diff
@@ -5,99 +5,39 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
-// NewSecretScopesAPI creates SecretScopesAPI instance from provider meta
-func NewSecretScopesAPI(ctx context.Context, m any) SecretScopesAPI {
-	return SecretScopesAPI{m.(*common.DatabricksClient), ctx}
-}
-
-// SecretScopesAPI exposes the Secret Scopes API
-type SecretScopesAPI struct {
-	client  *common.DatabricksClient
-	context context.Context
-}
-
-// SecretScopeList holds list of secret scopes
-type SecretScopeList struct {
-	Scopes []SecretScope `json:"scopes,omitempty"`
-}
-
 // SecretScope is a struct that encapsulates the secret scope
 type SecretScope struct {
-	Name                   string            `json:"name" tf:"force_new"`
-	BackendType            string            `json:"backend_type,omitempty" tf:"computed"`
-	InitialManagePrincipal string            `json:"initial_manage_principal,omitempty" tf:"force_new"`
-	KeyvaultMetadata       *KeyvaultMetadata `json:"keyvault_metadata,omitempty" tf:"force_new"`
-}
-
-// KeyvaultMetadata Azure Key Vault metadata wrapper
-type KeyvaultMetadata struct {
-	// /subscriptions/.../resourceGroups/.../providers/Microsoft.KeyVault/vaults/my-azure-kv
-	ResourceID string `json:"resource_id"`
-	// https://my-azure-kv.vault.azure.net/
-	DNSName string `json:"dns_name"`
-}
-
-type secretScopeRequest struct {
-	Scope                  string            `json:"scope,omitempty"`
-	BackendType            string            `json:"scope_backend_type,omitempty"`
-	InitialManagePrincipal string            `json:"initial_manage_principal,omitempty"`
-	BackendAzureKeyvault   *KeyvaultMetadata `json:"backend_azure_keyvault,omitempty"`
-}
-
-// Create creates a new secret scope
-func (a SecretScopesAPI) Create(s SecretScope) error {
-	req := secretScopeRequest{
-		Scope:                  s.Name,
-		InitialManagePrincipal: s.InitialManagePrincipal,
-		BackendType:            "DATABRICKS",
-	}
-	if s.KeyvaultMetadata != nil {
-		err := a.client.Config.EnsureResolved()
-		if err != nil {
-			return err
-		}
-		if !a.client.IsAzure() {
-			//lint:ignore ST1005 Azure is a valid capitalized string
-			return fmt.Errorf("Azure KeyVault is not available")
-		}
-		req.BackendType = "AZURE_KEYVAULT"
-		req.BackendAzureKeyvault = s.KeyvaultMetadata
-	}
-	return a.client.Post(a.context, "/secrets/scopes/create", req, nil)
-}
-
-// Delete deletes a secret scope
-func (a SecretScopesAPI) Delete(scope string) error {
-	return a.client.Post(a.context, "/secrets/scopes/delete", map[string]string{
-		"scope": scope,
-	}, nil)
-}
-
-// List lists all secret scopes available in the workspace
-func (a SecretScopesAPI) List() ([]SecretScope, error) {
-	var listSecretScopesResponse SecretScopeList
-	err := a.client.Get(a.context, "/secrets/scopes/list", nil, &listSecretScopesResponse)
-	return listSecretScopesResponse.Scopes, err
+	Name                   string                                      `json:"name" tf:"force_new"`
```

Comment:
I think this is fine, but did you consider doing something like
```
type SecretScope workspace.CreateScope

func (s SecretScope) CustomizeSchema(m map[string]string) map[string]string {
  s["name"].ValidateFunc = validScope
  common.CustomizeSchemaPath(m, "backend_type").SetComputed()
  common.CustomizeSchemaPath(m, "initial_manage_principal").SetForceNew()
  common.CustomizeSchemaPath(m, "keyvault_metadata").SetForceNew()
}

func (s SecretScope) Aliases() map[string]string {
  return map[string]string{
    "scope": "name",
    "backend_azure_keyvault": "keyvault_metadata",
  }
}
```

This way, you don't need to adapt one structure into another, and your main structure is based off of the Go SDK, so it inherits any changes from the Go SDK automatically.

Reply:
Hmmm, I can't make it work - I'm getting empty values.